gateway/ln-gateway/src/lib.rs

```diff
@@ -419,113 +425,148 @@ impl Gateway {
+                            warn!("Disconnected from Lightning Node. Waiting {HTLC_STREAM_RETRY_SECONDS} seconds and trying again");
+                            sleep(Duration::from_secs(HTLC_STREAM_RETRY_SECONDS)).await;
+                        }
+                        HandleStreamResult::NoRetry => break,
                     }
                 }
+            },
+        );
+    }
+
+    async fn route_htlcs<'a>(
+        &'a self,
+        handle: &TaskHandle,
+        stream: RouteHtlcStream<'a>,
+        ln_client: Arc<dyn ILnRpcClient>,
+    ) -> HandleStreamResult {
+        let (lightning_public_key, lightning_alias, lightning_network) =
+            match ln_client.parsed_node_info().await {
+                Ok((
+                    lightning_public_key,
+                    lightning_alias,
+                    lightning_network,
+                    _block_height,
+                    _synced_to_chain,
+                )) => (lightning_public_key, lightning_alias, lightning_network),
+                Err(e) => {
+                    warn!("Failed to retrieve Lightning info: {e:?}");
+                    return HandleStreamResult::RetryAfterDelay;
+                }
+            };
 
-                self_copy.handle_disconnect(htlc_task_group).await;
+        let gateway_config = if let Some(config) = self.clone_gateway_config().await {
+            config
+        } else {
+            self.set_gateway_state(GatewayState::Configuring).await;
+            info!("Waiting for gateway to be configured...");
+            self.gateway_db
+                .wait_key_exists(&GatewayConfigurationKey)
+                .await
+        };
 
-                warn!("Disconnected from Lightning Node. Waiting 5 seconds and trying again");
-                sleep(Duration::from_secs(5)).await;
-            }
-        });
-    }
+        if gateway_config.network != lightning_network {
+            warn!(
+                "Lightning node does not match previously configured gateway network : ({:?})",
+                gateway_config.network
+            );
+            info!(
+                "Changing gateway network to match lightning node network : ({:?})",
+                lightning_network
+            );
+            self.handle_set_configuration_msg(SetConfigurationPayload {
+                password: None,
+                network: Some(lightning_network),
+                num_route_hints: None,
+                routing_fees: None,
+                per_federation_routing_fees: None,
+            })
+            .await
+            .expect("Failed to set gateway configuration");
+            return HandleStreamResult::ImmediatelyRetry;
+        }
+
+        let lightning_context = LightningContext {
+            lnrpc: ln_client,
+            lightning_public_key,
+            lightning_alias,
+            lightning_network,
+        };
+        self.set_gateway_state(GatewayState::Running { lightning_context })
+            .await;
```

Comment:
Can we add a `info!` here so we know when the gateway is running?

Reply:
Done.